rfd/0201-native-auto-approval.md

```diff
@@ -0,0 +1,369 @@
+configured trait. For example, given the following AMR, a user is pre-approved
+if they are any level "L1" or "L2" or "L3" and they are on the "Cloud" team and
+they are located in "Seattle". If the user is level "L1" and located in
+"Seattle", but they are on the "Tools" team, they would not be pre-approved.
+
+```yaml
+kind: access_monitoring_rule
+version: v1
+metadata:
+  name: cloud-dev-pre-approved
+spec:
+  notification:
+    name: teleport-slack
+    recipients: ["#dev-cloud"]
+  automatic_approval:
+    name: teleport-native
+    native:
+       level: ["L1", "L2", "L3"]
+       team: ["Cloud"]
+       location: ["Seattle"]
+  condition: |
+    contains_any(access_request.spec.roles, set("cloud-dev"))
+  subjects:
+    - access_request
+```
+
+### Native Plugin Deployment
+The native plugin is deployed just like all the other access request plugins. It
+will be hosted by Teleport, but will not be self-hostable. Teleport will start
+the native plugin instance during initialization. The native plugin will not
+support any user configuration options.
+
+### Web UI Access Monitoring Rules
+The Teleport Web UI will provide a more user friendly approach to configuring
+auto approvals. Users will now be able to navigate to the **Access Requests**
+page and configure auto approvals, similarly to how notification routing is
+configured.
+
+The `Create a New Access Monitoring Rule` form will now be used to configure
+both notification routing rules, as well as automatic approval rules. The
+available configuration input for automatic approvals will be toggled depending
+on the selected plugin name/type.
+
+![create-amr](assets/0201-create-amr.png)
+
+The submitted form will be converted into an AMR that would look like:
+
+```yaml
+kind: access_monitoring_rule
+version: v1
+metadata:
+  name: cloud-dev-pre-approved
+spec:
+  notification:
+    name: slack-default
+    recipients: ["#dev-cloud"]
+  automatic_approval:
+    name: access-native
+    native:
+       level: ["L1"]
+       team: ["Cloud"]
+       location: ["Seattle"]
+  condition: |
+    contains_any(access_request.spec.roles, set("cloud-dev"))
+  subjects:
+    - access_request
+```
+
+The Access Monitoring Rules overview page will be modified to display both
+notification rules, as well as automatic approval rules. This page will allow
+user to see a quick overview of the automatic approvals currently enabled. The
+overview simply displays the access monitoring rule name, plugin/integration
+name, and the roles that are automatically approved. Users will need to click
+on the **View** button to see the actual conditions for auto approval.
+
+![view-amr](assets/0201-view-amr.png)
+
+## Security
+Automatic approvals is already a supported feature, although it is currently
+only supported when integrated with an external incident management system. The
+same security concerns apply with built-in auto approvals, as they apply to auto
+approvals with an external plugin.
+
```

Comment:
Need section about product metrics and audit logging.

Reply:
We've been collecting metrics for access requests with Teleport Connect, but we don't collect metrics when not using Teleport Connect. Do you know if there was any reason we were not collecting these? Did we just never get around to it?

Reply:
Yep, just never got around to it but would be good to add them. They're basically the main Identity feature :)